WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -186,7 +198,15 @@ final class ProductFormViewModel: ProductFormViewModelProtocol {
     }
 
     func hasUnsavedChanges() -> Bool {
-        return product != originalProduct || productImageActionHandler.productImageStatuses.hasPendingUpload || password != originalPassword
+        guard isBackgroundImageUploadEnabled else {
+            return product != originalProduct || productImageActionHandler.productImageStatuses.hasPendingUpload || password != originalPassword
+        }
+        let hasProductChanges = product.product.copy(images: []) != originalProduct.product.copy(images: [])
```

Comment:
Nit: What do you think about naming it like `hasAnythingApartFromImagesChangedInTheProduct` or something like that to explain why we are stripping images from it? 
I feel that the name I proposed isn't good either. We can add comments if that is cleaner. 😄

Reply:
Good point! Updated to `hasProductChangesExcludingImages` in https://github.com/woocommerce/woocommerce-ios/pull/7048/commits/9dd3efde0d2f70d710601cd8abf7ea90f30a8685